### ISSUE TITLE
Crash in AccountService syncBlogsForAccount:success:failure: getting nil account on signout

### ISSUE THREAD

**Issue by astralbodies** [wordpress-mobile/WordPress-iOS#2607]
When signing out sometimes AccountService is triggered with the nil account causing a crash because of the inability to fulfill a fault.

```
2014-10-17 12:39:59:846 WordPress[1465:807] Successfully unregistered device ID 6993762
2014-10-17 12:39:59.935 WordPress[1465:430535] *** Terminating app due to uncaught exception 'NSObjectInaccessibleException', reason: 'CoreData could not fulfill a fault for '0xd0000000017c0008 <x-coredata://3CD49754-5095-4767-B362-CCD237CFDD28/Blog/p95>''
*** First throw call stack:
(0x182dfa084 0x1934380e4 0x182ac65c4 0x182ac5a48 0x182ad7330 0x182ad71d4 0x182ae1958 0x182ae134c 0x183bee414 0x183bedf34 0x183bd74c4 0x182ad8e90 0x182ad7248 0x100188440 0x100187414 0x182b4d270 0x1013e4ee0 0x1013ef8e8 0x1013e8210 0x1013f175c 0x1013f2f18 0x193c592e4 0x193c58fa8)
libc++abi.dylib: terminating with uncaught exception of type _NSCoreDataException
```

**Comment by koke** [comment]
Maybe fixed by #3896 ?

**Comment by rachelmcr** [comment]
I don't see any reports of this crash after July, so it looks like #3896 fixed it. If it turns out this isn't entirely fixed, please reopen.
